staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -268,7 +271,15 @@ func NewConfig(codecs serializer.CodecFactory) *Config {
 		// on the size increase the "copy" operations in a json patch
 		// can cause.  See
 		// https://github.com/etcd-io/etcd/blob/release-3.3/etcdserver/server.go#L90.
-		JSONPatchMaxCopyBytes:        int64(10 * 1024 * 1024),
+		JSONPatchMaxCopyBytes: int64(10 * 1024 * 1024),
+		// 10MB is the recommended maximum client request size in bytes.
+		// See
+		// https://github.com/etcd-io/etcd/blob/release-3.3/etcdserver/server.go#L90.
+		// A request body might be encoded in json, and is converted to
+		// proto when persisted in etcd. Assuming the upper bound of
+		// the size ratio is 10:1, we set 100MB as the largest request
```

Comment:
Kudos to @jpbetz, he looked into a production cluster, the biggest json:proto ratio for a single object is ~2.3. So 10 is a very generous estimation.

FWIW, I also did some maths in mind:
* large primitive type Kubernetes have is int64, size ratio for int64 is log(2^64-1)/log10/2 = 9.1
* for field names, the longest field that is wrapped in a list is `terminationMessagePolicy`, which is 24 chars long. It will be 1 byte in proto. ([bash one-liner](https://pastebin.com/AqRyprHi)).